src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1387,17 +1403,27 @@ private function dumpValue($value, $interpolate = true)
 
             return sprintf('array(%s)', implode(', ', $code));
         } elseif ($value instanceof IteratorArgument) {
+            $countCode = array();
+            $countCode[] = 'function() {';
+            $operands = array(0);
+
             $code = array();
             $code[] = 'new RewindableGenerator(function() {';
             foreach ($value->getValues() as $k => $v) {
-                $v = $this->wrapServiceConditionals($v, sprintf("        yield %s => %s;\n", $this->dumpValue($k, $interpolate), $this->dumpValue($v, $interpolate)));
-                foreach (explode("\n", $v) as $v) {
-                    if ($v) {
-                        $code[] = '    '.$v;
+                $y = $this->wrapServiceConditionals($v, sprintf("        yield %s => %s;\n", $this->dumpValue($k, $interpolate), $this->dumpValue($v, $interpolate)));
+                foreach (explode("\n", $y) as $y) {
+                    if ($y) {
+                        $code[] = '    '.$y;
                     }
                 }
+
+                ($c = $this->getServiceConditionals($v)) ? $operands[] = "(int) ($c)" : ++$operands[0];
```

Comment:
can you move this line before calling wrapServiceConditionals and remove the need for the `$y` variable?